### ISSUE TITLE
[no-cycle] regression between v4.15.2 and 4.16.0 causing false positives

### ISSUE THREAD

**Issue by Zamiell** [un-ts/eslint-plugin-import-x#397]
After upgrading my version of `eslint-plugin-import-x`, my project fails lint with a false positive.

Steps to reproduce:

```sh
git clone git@github.com:complete-ts/complete.git
cd complete
git checkout 2231fd46e555eb889a5f97c6635ade98383e88ad
npm ci
npx eslint ./scripts/update.ts
```

If you checkout any previous commit (and reinstall dependencies), then linting this file will pass, proving that it is a regression. But moreover, the error is obviously a false positive, since it seems impossible to have a cycle that comes from an external dependency. (It's a monorepo package, so not *technically* an external dependency, but it is basically external as far as the "update.ts" file is concerned.)

**Comment by JounQin** [comment]
Please help to test:

```sh
npm i https://pkg.pr.new/eslint-plugin-import-x@399
```

**Comment by Zamiell** [comment]
looks fixed, thanks for the quick turnaround!
